contrib/IBM/sarama.v1/sarama.go

```diff
@@ -74,6 +74,7 @@ func WrapPartitionConsumer(pc sarama.PartitionConsumer, opts ...Option) sarama.P
 			// kafka supports headers, so try to extract a span context
 			carrier := NewConsumerMessageCarrier(msg)
 			if spanctx, err := tracer.Extract(carrier); err == nil {
+				opts = append(opts, tracer.WithExtractedSpanLinks(spanctx))
 				opts = append(opts, tracer.ChildOf(spanctx))
```

Comment:
On line 77, you will append an uninitialized StartSpanOption in the case that `spanctx` does not contain any `spanLinks`. That's not what you want to do.
Instead, you should check whether the `spanCtx` contains spanLinks and, if so, append to the opts slice. 
Note: This applies to all places you've called tracer.WithExtractedSpanLinks, not just line 77 of this file.

By the way/ for future reference/ in case it's helpful:`append` can take multiple args, meaning you could do this `opts = append(opts, tracer.ChildOf(spanctx), tracer.WithExtractedSpanLinks(spanctx))`